packages/protocol/contracts/dapis/RrpDapiServer.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "../AirnodeRrp.sol";
+import "./reducers/MeanMedianHybrid.sol";
+
+contract RrpDapiServer is MeanMedianHybrid {
+    struct Dapi {
+        uint256 noResponsesToReduce;
+        int256 toleranceInPercentages;
+        uint256 requesterIndex;
+        bytes32[] templateIds;
+        address[] designatedWallets;
+        address reduceAddress;
+        bytes4 reduceFunctionId;
+        mapping(uint256 => int256[]) requestIndexToResponses;
+        mapping(uint256 => uint256) requestIndexToResponsesLength;
```

Comment:
Before, we were using `.push()`, using `.length` for the array length and calling `delete` when we no longer needed it. Now, we're keeping a separate length variable for each dAPI request and incrementing/resetting it manually.

Reply:
Does it have to be as granular as `requestIndex`?

Won't dAPIs always require the same number of responses for aggregation? And this is a parameter in the `Dapi` struct.

Reply:
So if `noResponsesToReduce` for a dAPI is `7`, `requestIndexToResponses[dapiId][*].length` will always be `7`. Then, we need a separate counter to keep how many responses we have received because we no longer can use the response length as the counter as we did before. Refer to how it is used in `fulfill()`.

Reply:
https://github.com/api3dao/airnode/pull/303/files#r624727056